### PR TITLE
remove deprecated langchain imports and switch to langchain-huggingface

### PR DESCRIPTION
**Pull request by rbrugaro**
## Description
Remove deprecated langchain imports from community package and switch to langchain-huggingface

## Issues
n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.
- [ x] Bug fix (non-breaking change which fixes an issue)

## Dependencies
new dependencies:
langchain_huggingface

## Tests
validated with
langchain                 0.2.12 
langchain-huggingface     0.0.3

### CODE REVIEW

**Comment by rbrugaro** [comment]
@xuechendi could you please review or tag who should review this? Thx!

File: comps/embeddings/langchain/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 docarray[full]
 fastapi
 huggingface_hub
-langchain
+langchain==0.2.12
```
Comment (xuechendi) [inline comps/embeddings/langchain/requirements.txt:4]: why fix the version?

File: comps/embeddings/langchain/embedding_tei.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import os
 import time
 
-from langchain_community.embeddings import HuggingFaceHubEmbeddings
+from langchain_huggingface import HuggingFaceEndpointEmbeddings
```
Comment (xuechendi) [inline comps/embeddings/langchain/embedding_tei.py:7]: I see a lot of other components using "HuggingFaceHubEmbeddings" instead of "HuggingFaceEndpointEmbeddings". What is the motivation behind this change?

Comment (rbrugaro) [reply comps/embeddings/langchain/embedding_tei.py:7]: Huggingface is moving their langchain integrations into their partner package "langchain_huggingface" rather than the community library. Those in community are deprecated starting in langchain 0.3 https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/embeddings/huggingface_hub.py

File: comps/embeddings/langchain/requirements.txt (hunk 1 above)
Comment (rbrugaro) [reply comps/embeddings/langchain/requirements.txt:4]: I wanted to show test are passing with the latest langchain but we can remove the version constraint

File: comps/embeddings/langchain/embedding_tei.py (hunk 2 above)
Comment (xuechendi) [reply comps/embeddings/langchain/embedding_tei.py:7]: I see, Rita, good to know :)
@ftian1 @lvliang-intel @lkk12014402 @XuhuiRen , would like to know if you are planning to update the Embedding packages here. According to Rita, it makes sense to do massive replace all "HuggingFaceHubEmbeddings" to "HuggingFaceEndpointEmbeddings"

File: comps/embeddings/langchain/requirements.txt (hunk 1 above)
Comment (chensuyue) [reply comps/embeddings/langchain/requirements.txt:4]: let's remove the version limitation, we will freeze it in the release tag in our future release. And in the main branch, let's continue use the latest one.

File: comps/embeddings/langchain/embedding_tei.py (hunk 2 above)
Comment (ftian1) [reply comps/embeddings/langchain/embedding_tei.py:7]: yes, it should be the direction.

**Review by ftian1 — APPROVED** [review]

**Review by XuhuiRen — APPROVED** [review]
LGTM

File: comps/embeddings/langchain/requirements.txt (hunk 1 above)
Comment (rbrugaro) [reply comps/embeddings/langchain/requirements.txt]: ok, removed it

**Comment by rbrugaro** [comment]
all checks passed. plz approve @xuechendi

**Review by xuechendi — APPROVED** [review]
LGTM
